ui/browser/views/navbar/navbar.jsx

```diff
@@ -10,60 +10,76 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+      setLocation, bookmark, unbookmark, onLocationChange, onLocationContextMenu, onLocationReset
+    } = this.props;
 
-  return (
-    <div id="browser-navbar">
-      <Btn title="Menu" icon="bars fa-lg"
-        onClick={() => menuBrowser(dispatch)} />
+    if (page == null) {
+      return <div id="browser-navbar"></div>;
+    }
 
-      <a id="pages-button">
-        <span className="page-count">{pages.length}</span>
-        {"Pages"}
-      </a>
-      <Btn title="Back" icon="arrow-left fa-lg"
-        onClick={e => getCurrentWebView(e.target.ownerDocument).goBack()}
-        disabled={!page.canGoBack} />
-      <Btn title="Forward" icon="arrow-right fa-lg"
-        onClick={e => getCurrentWebView(e.target.ownerDocument).goForward()}
-        disabled={!page.canGoForward} />
-      <Btn title="Refresh" icon="refresh"
-        onClick={e => getCurrentWebView(e.target.ownerDocument).reload()}
-        disabled={!page.canRefresh} />
+    return (
+      <div id="browser-navbar">
+        <Btn title="Menu" icon="bars fa-lg"
+          onClick={openMenu} />
 
-      <Location {...{ page, ipcRenderer }} />
+        <a id="pages-button">
+          <span className="page-count">{pages.length}</span>
+          {"Pages"}
+        </a>
+        <Btn title="Back" icon="arrow-left fa-lg"
+          onClick={navBack}
+          disabled={!page.canGoBack} />
+        <Btn title="Forward" icon="arrow-right fa-lg"
+          onClick={navForward}
+          disabled={!page.canGoForward} />
+        <Btn title="Refresh" icon="refresh"
+          onClick={navRefresh}
+          disabled={!page.canRefresh} />
 
-      <Btn title="Home" icon="home fa-lg"
-        onClick={e => getCurrentWebView(e.target.ownerDocument).goToIndex(0)}
-        disabled={!page.canGoBack} />
+        <Location {
+          ...{page, ipcRenderer, bookmark, unbookmark, onLocationChange, onLocationContextMenu, onLocationReset}
+        } />
 
-      <Btn title="Minimize" icon="minus fa-lg"
-        onClick={minimize} />
-      <Btn title="Maximize" icon="square-o fa-lg"
-        onClick={maximize} />
-      <Btn title="Close" icon="times fa-lg"
-        onClick={close} />
-    </div>
-  );
+        <Btn title="Minimize" icon="minus fa-lg"
+          onClick={minimize} />
+        <Btn title="Maximize" icon="square-o fa-lg"
+          onClick={maximize} />
+        <Btn title="Close" icon="times fa-lg"
+          onClick={close} />
+      </div>
+    );
+  }
 };
 
 NavBar.propTypes = {
   page: PropTypes.object,
-  dispatch: PropTypes.func.isRequired,
   ipcRenderer: PropTypes.object.isRequired,
+  navBack: PropTypes.func.isRequired,
+  navForward: PropTypes.func.isRequired,
+  navRefresh: PropTypes.func.isRequired,
+  minimize: PropTypes.func.isRequired,
+  maximize: PropTypes.func.isRequired,
+  close: PropTypes.func.isRequired,
+  openMenu: PropTypes.func.isRequired,
+
+  // For <Location>
+  bookmark: PropTypes.func.isRequired,
```

Comment:
Passing these props all the way down the component hierarchy might get noisy / annoying (having to copy/paste the same thing in a number of places).  But, seems like the right direction to go in for testability and simplifying the components so let's go with it